lib/bundler/definition.rb

```diff
@@ -371,14 +381,8 @@ def ensure_equivalent_gemfile_and_lockfile(explicit_flag = false)
         end
       end
 
-      new_deps = @dependencies - @locked_deps
-      deleted_deps = @locked_deps - @dependencies
-
       added.concat new_deps.map {|d| "* #{pretty_dep(d)}" } if new_deps.any?
-
-      if deleted_deps.any?
-        deleted.concat deleted_deps.map {|d| "* #{pretty_dep(d)}" }
-      end
+      deleted.concat deleted_deps.map {|d| "* #{pretty_dep(d)}" } if deleted_deps.any?
```

Comment:
@allenzhao you can make a separate refactor commit removing those conditionals. There's no harm in concatenating an empty array (if `deleted_deps` was empty).